web/src/presence.ts

```diff
@@ -110,18 +107,22 @@ export function status_from_raw(raw: RawPresence): PresenceStatus {
         timestamp for idle).
     */
     if (age(active_timestamp) < offline_threshold_secs) {
+        last_active = active_timestamp;
         return {
             status: "active",
             last_active,
         };
     }
 
     if (age(idle_timestamp) < offline_threshold_secs) {
+        last_active = idle_timestamp;
         return {
             status: "idle",
             last_active,
         };
     }
+    
+    last_active = active_timestamp;
```

Comment:
Can you add a comment here summarizing the reasoning from the design thread?

Reply:
Comment added.